File: android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/FlutterChannel.kt
package com.wrbug.gitbbs.flutterbridge

import android.util.Log
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

abstract class FlutterChannel {
    companion object {
        /// 修改以下参数，务必将flutter和iOS相应值也修改， 建议保持默认
        /// iOS ： ios/Runner/FlutterBridge/FlutterChannel.swift
        /// flutter：lib/constant/NativeBridgeConstant.dart
        private const val prefix = "com.wrbug.gitbbs"
    }

    private val handler = MethodChannel.MethodCallHandler { p0, p1 ->
        dispatch(p0, p1)
    }

    fun inject(messenger: BinaryMessenger) {
        MethodChannel(messenger, prefix + "/" + getChannel()).setMethodCallHandler(handler)
    }

    abstract fun getChannel(): String
    abstract fun dispatch(call: MethodCall, result: MethodChannel.Result)
}

File: ios/Runner/FlutterBridge/FlutterChannel.swift
//
//  FlutterChannel.swift
//  Runner
//
//  Created by WrBug on 2019/3/7.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation
import Flutter

/// 修改以下参数，务必将flutter和iOS相应值也修改， 建议保持默认
/// android ： android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/FlutterChannel.kt
/// flutter：lib/constant/NativeBridgeConstant.dart
let prefix:String="com.wrbug.gitbbs";
class FlutterChannel {
    
    
    func handler() -> FlutterMethodCallHandler{
        return {(call, result ) in
            self.dispatch(call , result);
        }
    }
    func inject(controller:FlutterViewController)  {
        FlutterMethodChannel.init(name: prefix+"/"+getChannel(),binaryMessenger:controller)
        .setMethodCallHandler(handler())
    }
    func  dispatch(_ call:  FlutterMethodCall,_ result: FlutterResult){}
    
    func getChannel() -> String {
        return "";
    }
}
